Scripts/Source/minai_Sex.psc

```diff
@@ -1194,3 +1033,278 @@ string Function GetFactionsForActor(actor akTarget)
+      ; some actions in ostim prevents actors from talking, which makes sense
+    elseif(currSex == 0)
+      maleActors = PapyrusUtil.PushActor(maleActors, currActor)
+    elseif(currSex == 1)
+      femaleActors = PapyrusUtil.PushActor(femaleActors, currActor)
+    endif
+  endwhile
+
+  ; if no female npc, just pick random male. Can be none though if it's player only scene
+  if(femaleActors.length == 0)
+    return getRandomActor(maleActors)
+  endif
+
+  ; if no male npc, just pick random female. Can be none though if it's player only scene
+  if(maleActors.length == 0)
+    return getRandomActor(femaleActors)
+  endif
+
+  ; pick weighted type of npc who will talk
+  bool isFemale = Utility.RandomInt(1, 100) <= commentFemaleWeight
+
+  if(isFemale)
+    return getRandomActor(femaleActors)
+  else
+    return getRandomActor(maleActors)
+  endif
+EndFunction
+
+function setPrevSpeed(int ThreadID, int prevSpeed)
+  JMap.setInt(threadsPrevSpeedsMap, ThreadID, prevSpeed)
+endfunction
+
+function resetPrevSpeed(int ThreadID)
+  JMap.setInt(threadsPrevSpeedsMap, ThreadID, -1)
+endfunction
+
+int function getPrevSpeed(int ThreadID)
+  return JMap.getInt(threadsPrevSpeedsMap, ThreadID)
+endfunction
+
+function removePrevSpeed(int ThreadID)
+  JMap.removeKey(threadsPrevSpeedsMap, ThreadID)
+endfunction
+
+string function buildSceneString(string sceneId, string sceneTagsString, string actionString)
+  string result = sceneId+"."
+  if(sceneTagsString != "")
+    result += " Scene can be described with this tags: " + sceneTagsString + "."
+  endif
+  if(actionString != "")
+    result += " These actions happen in a scene: " + actionString
+  endif
+  return result
+endfunction
+
+string function buildSceneFallbackDescription(int ThreadID, string framework, string eventType)
+  string sceneId = ""
+  string actorString = ""
+  string actionString = ""
+  string sceneTagsString = ""
+  if(framework == "ostim")
+    sceneId = OThread.GetScene(ThreadID)
+    string[] actionTypes = OMetadata.GetActionTypes(sceneId)
+    string[] sceneTags = OMetadata.GetSceneTags(sceneId)
+    Actor[] actors = OThread.GetActors(ThreadID)
+    actorString = JoinActorArray(actors, ", ")
+    actionString = JoinStringArray(actionTypes, ", ")
+    sceneTagsString = JoinStringArray(sceneTags, ", ")
+  else
+    sslThreadController controller = slf.GetController(ThreadID)
+    string[] sceneTags= controller.Animation.GetRawTags()
+    sceneId = controller.Animation.Name
+    Actor[] actors = slf.GetController(ThreadID).Positions
+    actorString = JoinActorArray(actors, ", ")
+    sceneTagsString = JoinStringArray(sceneTags, ", ")
+    int count = actors.length
+    while(count > 0)
+      count -= 1
+      string stageDescription = GetSexStageDescription(controller.Animation.FetchStage(controller.Stage)[0])
+
+      if(stageDescription != "")
+        actionString += GetActorNameForSex(actors[count])+" is "+ stageDescription
```

Comment:
This will have the same description for either actor, right? That's not what should happen.

Reply:
my bad, changed `0` to `count`(if I understood logic here correctly), kind of obsolete if we will switch to new description approach